benchmarks/llm/prepare.py

```diff
@@ -16,10 +21,35 @@ class Arguments:
     config: str = None
 
 
+@dataclass
+class ModelArgs(llama.model.ModelArgs):
+    use_scaled_rope: bool = True
+
 class MyParser(TuneCLIParser):
     def parse_args(self, args=None) -> argparse.Namespace:
         """Parse CLI arguments"""
-        return self._parser.parse_args(args)
+        parsed_args = self._parser.parse_args(args)
+        # Workaround to send a list to of ignore_patterns as self._parser does
+        # not support a list in input
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "--ignore-patterns",
+            type=str,
+            action='append',
+        )
+        ignore_patterns_args, _ = parser.parse_known_args(args)
+        if ignore_patterns_args.ignore_patterns:
+            parsed_args.ignore_patterns = ignore_patterns_args.ignore_patterns
+        return parsed_args
+
+
+def generate_model(model_parallel_size, params_path:Path):
+    params = json.loads(params_path.read_text())
+    torch.distributed.init_process_group(rank=0, world_size=1)
+    fairscale.nn.model_parallel.initialize.initialize_model_parallel(model_parallel_size)
+    for i in range(model_parallel_size):
+        model = llama.model.Transformer(ModelArgs(**params))
+        torch.save(model.state_dict(), params_path.with_name(f"consolidated.{i:02}.pth"))
```

Comment:
should work for llama_8b but produces a 30GB checkpoint instead of a 16GB checkpoint. Need to check how to make this work for 70B and reduce the size if possible